### DISCUSSION TITLE
Antique Car Stereo Bluetooth Audio Retrofit with ESP32

### DISCUSSION THREAD

**Discussion by jordancrubin** [pschatzmann/ESP32-A2DP#198] [Show and tell]
I'd picked up another Sapphire 9 radio, this time its only a repair, not a restoration, left in the original patina, and it seems in good condition.  I was however asked if an auto-switching Bluetooth module could be designed for such a unit.  Not one with a switch but automatic like a modern radio. This took some thinking.

Id decided to go with an ESP32 as a platform for this using a PCM5102 DAC as a compliment.  I'd quickly found 2 areas for possible install, but first a design would need to be established.

A library from Phil Schatzmann exists in the community to provide an easy interface between the bluetooth and the DAC.  A proof of concept is up and running quickly, I try it out on the Dynaco ST-70 and its working good.

Id chosen the voltage regulator as the voltage can be specifically selected, but their implementation was crap, and it was difficult to work with, I suffered through a few of them before cutting of the pot before figuring out the trace cut.

A power budget is then taken, i find it way below budget so I'm not concerned.

When more parts arrive well continue in chapter 2.

[https://youtu.be/goUFxiTSTvU](https://youtu.be/goUFxiTSTvU)

**Comment by jordancrubin** [comment]
With the parts needed to continue we can now move on with the repair of the radio itself, aw well as a cleaning and service of the unit, however, a restoration is not authorized.  

With the radio repaired and tested a good point for power pickup is determined. It will be switched power so that the ESP32 and associated components turns on and off with the radio.  In the mean time i've been soaking the ground lug which was completely rusted, eventually it broke off.  I dropped a new bolt of the same size right in.

The 5 volt regulator was connected into the switched power of the radio and tested good. We can use the radio as a power source going forward.

The mux is then designed into the circuit so that the AM and Bluetooth signals can auto switch into the radios pre-amp input.  The mux switching is controlled by the ESP32.

To make things more interesting, Ive digitized my voice into the esp32 when the Bluetooth connects and disconnects.  After testing out the voice an analysis is done to determine the best point to inject the signals back into the radio. Schematic point 13 is determined.

Wiring in both AM and Bluetooth into the mux and the output to point 13 allows for a full running auto switching system working as designed.  

The next chapter will deal with fitting all of this into the unit. 

[https://www.youtube.com/watch?v=mIX057mQHOA](https://www.youtube.com/watch?v=mIX057mQHOA)

**Comment by BlueSteelTuner** [comment]
First time on GitHub, so apologies if I overstep.
I'm interested in creating Bluetooth for my wife's car, a 2008 Saab 9-3 2.0 SC b207r 5spd automatic.  She likes the original look of the head unit, but would like to listen to her Spotify via cell.
Your project here appears similar?
What would you suggest?
Any input greatly appreciated.

**Reply by jordancrubin** [reply]
Of course you could go a similar route, keeping in mind however that the AM radio's of the 60's are vastly different than a late model Saab.  There would be no point to point, all IC integration, and little room to work.  Such modifications on early 80's delorean head units are challenging.  2008 would be something...

**Reply by BlueSteelTuner** [reply]
OK, thx.  I've never seen a video like yours; very special.
